module/VuFind/src/VuFind/Http/CachingDownloader.php

```diff
@@ -0,0 +1,193 @@
+<?php
+/**
+ * Caching downloader.
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Villanova University 2022.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Http
+ * @author   Mario Trojan <mario.trojan@uni-tuebingen.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+namespace VuFind\Http;
+
+use Laminas\Cache\Storage\StorageInterface;
+use VuFind\Cache\Manager as CacheManager;
+use VuFind\Exception\HttpDownloadException;
+
+/**
+ * Caching downloader.
+ *
+ * @category VuFind
+ * @package  Http
+ * @author   Mario Trojan <mario.trojan@uni-tuebingen.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class CachingDownloader implements \VuFindHttp\HttpServiceAwareInterface
+{
+    use \VuFindHttp\HttpServiceAwareTrait;
+
+    /**
+     * CacheManager to update caches if necessary.
+     *
+     * @var CacheManager
+     */
+    protected $cacheManager;
+
+    /**
+     * Stored client options for cache key generation.
+     *
+     * @var array
+     */
+    protected $cacheOptions = [];
+
+    /**
+     * Cache to use for downloads
+     *
+     * @var StorageInterface
+     */
+    protected $cache;
+
+    /**
+     * Cache ID to use for downloads
+     *
+     * @var string
+     */
+    protected $cacheId;
+
+    /**
+     * Constructor
+     *
+     * @param string $cacheManager Base directory for cache
+     */
+    public function __construct(CacheManager $cacheManager)
+    {
+        $this->cacheManager = $cacheManager;
+        $this->setUpCache('downloader');
+    }
+
+    /**
+     * Get cache and initialize it, if necessary.
+     *
+     * @return StorageInterface
+     */
+    protected function getCache()
+    {
+        if ($this->cache == null) {
+            $cacheName = $this->cacheManager->addDownloaderCache($this->cacheId);
```

Comment:
Since the name isn't really needed anywhere, would it make sense to name the method `getDownloaderCache` and make it return the cache directly?

Reply:
Not sure if i understand correctly => Do you mean the method in VuFind\Cache\CacheManager itself? I tried to keep the name + usage similar to other existing methods, especially `addLanguageCacheForTheme`.

Reply:
That's what I meant, but following the existing pattern here makes sense.